src/input/type_base.cc

```diff
@@ -109,6 +110,20 @@ void TypeBase::add_attribute(std::string name, json_string val) {
 
 
 void TypeBase::write_attributes(ostream& stream) const {
+	// print hash of generic type and parameters into separate keys
+	if (generic_type_.size()) {
```

Comment:
Remove this method completely even from unit_tests, use the fixture class instead. 
Touch the test: type_base_test.cpp